Framework/DataHandling/src/MultiPeriodLoadMuonStrategy.cpp

```diff
@@ -0,0 +1,138 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidDataHandling/MultiPeriodLoadMuonStrategy.h"
+#include "MantidAPI/Run.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidDataHandling/LoadMuonNexusV2Helper.h"
+#include "MantidDataObjects/TableWorkspace.h"
+#include "MantidDataObjects/Workspace2D.h"
+
+namespace Mantid {
+namespace DataHandling {
+
+using namespace API;
+using namespace NeXus;
+using namespace DataObjects;
+
+namespace {
+std::string GOODFRAMESPROP{"goodfrm"};
+constexpr bool MULTIPERIODSLOADED = true;
+} // namespace
+
+// Constructor
+MultiPeriodLoadMuonStrategy::MultiPeriodLoadMuonStrategy(
+    Kernel::Logger &g_log, const std::string filename, NXEntry &entry,
```

Comment:
If you make `filename` a reference here, but leave it as a "normal" type in the header the compiler will do the right thing rather than a potential double copy

Reply:
Again, wondering whether passing by reference is best here. Also not sure what you mean by normal type.

Reply:
Your right, it's me being overzealous with tweaks and changes